includes/derivatives.inc

```diff
@@ -116,11 +116,19 @@ function islandora_ocr_get_uploaded_file(AbstractDatastream $datastream) {
  *   The page datastream that the derivatives will be generated for.
  * @param array $options
  *   The options for tesseract.
+ * @param bool $ingest
+ *   Whether or not to ingest the datastreams during the execution of this
+ *   function and then clean up the generated OCR files from the temp folder;
+ *   otherwise, references will be returned to the converted files, and
+ *   generated files will have to be cleaned up later.
  *
- * @return bool
- *   TRUE on success, FALSE otherwise.
+ * @return bool|array
+ *   If $ingest is true, this returns TRUE on success and FALSE on failure. If
+ *   $ingest is FALSE, this returns an associative array containing 'OCR' and
+ *   'HOCR', each paired with either the path to their respective file, or FALSE
+ *   if generation of that datastream had failed.
  */
-function islandora_ocr_derive_datastreams(AbstractDatastream $datastream, array $options) {
+function islandora_ocr_derive_datastreams(AbstractDatastream $datastream, array $options, $ingest = TRUE) {
```

Comment:
It almost feels like the case with `$ingest = FALSE` should be pulled out into a separate function which this functions then calls... Then instead of a separate parameter, you would just call this "internal" function? Maybe? Dunno.